pkg/datastore/ingest.go

```diff
@@ -92,7 +92,7 @@ func (s *Datastore) Ingest(ctx context.Context, datasetID string, content []byte
 	ingestionFlow.FillDefaults(filetype, opts.TextSplitterOpts)
 
 	// Mandatory Transformation: Add filename to metadata
-	em := &transformers.ExtraMetadata{Metadata: map[string]any{"filename": filename}}
+	em := &transformers.ExtraMetadata{Metadata: map[string]any{"filename": filename, "absPath": opts.FileMetadata.AbsolutePath}}
```

Comment:
Note: abspath may not exist if the content is coming via the API (server mode)

Reply:
Hmm yeah, good point. At some point I just want to identify the file. I guess in server mode we can just use the name because you can't upload two files with the same name :)